app/views/datasets/_contact.html.erb

```diff
@@ -29,14 +30,13 @@
       </h3>
 
       <p>
-        <span><%= foi_name_for(dataset) %></span>
-        <span><%= mail_to(foi_email_for(dataset), nil, {class: 'ga-contact'}) %></span>
-        <span>
-          <%= link_to(foi_web_address_for(dataset),
-              foi_web_address_for(dataset),
-              {class: 'ga-contact'})
-          %>
-        </span>
+        <%= content_tag(:span, foi_name_for(dataset)) if foi_name_for(dataset) != "" %>
+        <%= content_tag(:span, mail_to(foi_email_for(dataset), nil, {class: 'ga-contact'})) if foi_email_for(dataset) != "" %>
+        <%=
```

Comment:
You can pass a block to the `content_tag` method which would be a good thing to do here:

```erb
<% if foi_web_address_for(dataset).present? %>
  <%= content_tag(:span) do %>
    <%= link_to(foi_web_address_for(dataset), foi_web_address_for(dataset), { class: 'ga-contact'} ) %>
  <% end %>
<% end %>
```